ivy.el

```diff
@@ -914,6 +914,20 @@ When this directory doesn't exist, return nil."
   "When non-nil, `ivy-partial-or-done' should insert a space."
   :type 'boolean)
 
+(defun ivy-complete-or-done()
+  "Complete the minibuffer text with selected.
+if selected text is directory, expand directory,
+and if the text has been completed, forward to `ivy-alt-done' when second call."
+  (interactive)
+  (ivy-partial-or-done)
+  (when (not (eq this-command last-command))
+    (delete-minibuffer-contents)
+    (let ((ivy-text (ivy-state-current ivy-last)) dir)
```

Comment:
Please place the `dir` declaration on a separate line, so that it is not obfuscated.